torch/fx/node.py

```diff
@@ -69,11 +69,14 @@
     ]
 )
 
-_side_effectful_need_to_be_preserved_pre_dispatch: set[Callable] = {
+# Dynamo is unable to trace global set[Callable].__contains__.
+# See https://github.com/pytorch/pytorch/issues/145761. Since we only have
+# a handful of ops so switch to list of callables.
+_side_effectful_need_to_be_preserved_pre_dispatch: list[Callable] = [
```

Comment:
Why is dynamo tracing this line?

Reply:
in non-strict export, while_loop calls torch.compile, which inlines into PreDispatchTorchFunctionMode and reaches the `if op in _side_effectful_need_to_be_preserved_pre_dispatch` line.